### PR TITLE
fix some golint failures in pkg/registry/...

### PR DESCRIPTION
**Pull request by SataQiu**
**What type of PR is this?**
 /kind cleanup

**What this PR does / why we need it**:
Fix golint failures on :
- pkg/registry/apps/daemonset
- pkg/registry/apps/deployment
- pkg/registry/apps/statefulset
- pkg/registry/autoscaling/horizontalpodautoscaler
- pkg/registry/batch/cronjob
- pkg/registry/core/resourcequota
- pkg/registry/policy/poddisruptionbudget

**Which issue(s) this PR fixes**:
Refs #68026

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by SataQiu** [comment]
@/assign @liggitt

File: pkg/registry/apps/daemonset/strategy.go [hunk 1]
```diff
@@ -166,6 +166,7 @@ type daemonSetStatusStrategy struct {
 	daemonSetStrategy
 }
 
+// StatusStrategy is the default logic invoked when updating service status.
```
Comment (liggitt) [inline pkg/registry/apps/daemonset/strategy.go]: copy/paste error - comments refer to "service status", can just change to "object status" (applies to all)

Comment (SataQiu) [reply pkg/registry/apps/daemonset/strategy.go]: Thank you for your review!

**Pull request merged**